### ISSUE TITLE
[BUG] One hitting enemies by click is triggering the update of the achievement lists

### ISSUE THREAD

**Issue by erzads** [pokeclicker/pokeclicker#3228]
**Info**
 - OS: [Android 12
 - Browser [Chrome]
 - Version [e.g. `v0.10.0`]

**Description:**
Whenever you defeat a Pokemon and the game loads a new one, if your achievements list is sorted by and value other than Default, there is a call to update every achievement in the list.
This is specially noticeable when one-hitting enemies very fast.

**Steps to Reproduce:**
1. Sort the achivements list by progress
2. Defeat pokemon on a route you can one-hit them by click attack
3. Profile the application and notice the calls to update the achivements list.

**Expected Behavior:**
The achievements list should have a rate limit even when sorted.

**Additional Information:**
![Screenshot from 2022-10-21 10-24-05](https://user-images.githubusercontent.com/699344/197407838-3027b0dc-bccd-45fd-a183-d8a046b900d3.png)
![Screenshot from 2022-10-21 10-25-23](https://user-images.githubusercontent.com/699344/197407844-faa89beb-2a3b-4276-a545-1aed0e1472b6.png)
![Screenshot from 2022-10-21 10-25-36](https://user-images.githubusercontent.com/699344/197407849-798efee5-d84c-42a5-a5f7-5138bf30e239.png)

**Comment by jlin98** [comment]
Was experiencing some performance chug myself when I noticed this bug and the related pull request. I see how rate limiting the get sorted list calc would help, but wouldn't performance be even more better if the achievement modal only updated its page when active instead of all the time?

**Comment by erzads** [comment]
@jlin98 I don't believe it would work, otherwise you wouldn't receive a notification when you completed an achievement.

But I do agree it could update less often. I am not so sure how to implement a solution tough.

**Comment by jlin98** [comment]
I'm not talking about the progress update of each achievement (which should happen every game tick) but the update of the [achievement modal itself](https://github.com/pokeclicker/pokeclicker/blob/develop/src/components/achievementsModal.html). Right now it looks like it calls getAchievementListWithIndex to update which achievements it displays, and this happen every time an achievement progresses. But if the player isn't looking at the achievement modal, then there's no reason to update the display, especially since the entire (filtered) list needs to get sorted from scratch anyways.

Basically, I don't think there's a great reason to constantly recalculate what a ui element looks like when we know for sure no one is looking at it. Though I might be misunderstanding what achivementsModal is doing.

**Comment by erzads** [comment]
My bad, you might be right.
